source/server/admin/config_dump_handler.cc

```diff
@@ -384,10 +384,12 @@ void ConfigDumpHandler::addLbEndpoint(
   endpoint.set_hostname(host->hostname());
   Network::Utility::addressToProtobufAddress(*host->address(), *endpoint.mutable_address());
   if (host->addressListOrNull() != nullptr) {
-    for (auto& additional_address : *host->addressListOrNull()) {
-      auto& new_address = *endpoint.mutable_additional_addresses()->Add();
-      Network::Utility::addressToProtobufAddress(*additional_address,
-                                                 *new_address.mutable_address());
+    const auto& address_list = *host->addressListOrNull();
+    if (address_list.size() > 1) {
+      for (auto it = std::next(address_list.begin()); it != address_list.end(); ++it) {
+        auto& new_address = *endpoint.mutable_additional_addresses()->Add();
+        Network::Utility::addressToProtobufAddress(**it, *new_address.mutable_address());
```

Comment:
I'm assuming that `*it` is always non-null, but not sure why. Any insights?

Reply:
*it is always non-null as long as "it" is a valid iterator. We can assume that *it is always a valid reference (due to the check on len I have before the loop) and because we stop the loop before the iterator becomes invalid (it != address_list.end() check)

Reply:
Not sure I understand why `*it` will always refer to something non-null by this description.
My question was if a shared_ptr element in the vector (IIUC `address_list` is which is a vector of shared-ptrs) could be null, but I guess that it the use of shared-ptrs ensures that each element that was added to the vector will stay valid.